contracts/Core/CollectionManager.sol

```diff
@@ -81,46 +82,35 @@ contract AssetManager is AssetStorage, StateManager, AssetManagerParams, IAssetM
 
     function setCollectionStatus(bool assetStatus, uint16 id)
         external
-        onlyRole(ASSET_MODIFIER_ROLE)
+        onlyRole(COLLECTION_MODIFIER_ROLE)
         checkState(State.Confirm, epochLength)
     {
         require(id != 0, "ID cannot be 0");
-        require(collections[id].id == id, "Asset is not a collection");
+        require(id <= numCollections, "ID does not exist");
 
```

Comment:
require for the assetStatus to be opposite of collections[id].active. allows to remove updateregistry from if and reduces 1 if-else block